AK/FixedArray.h

```diff
@@ -15,40 +17,60 @@ namespace AK {
 template<typename T>
 class FixedArray {
 public:
-    FixedArray() = default;
-    explicit FixedArray(size_t size)
-        : m_size(size)
+    explicit FixedArray() = default;
+
+    static ErrorOr<FixedArray<T>> try_create(size_t size)
     {
-        if (m_size != 0) {
-            m_elements = static_cast<T*>(kmalloc_array(m_size, sizeof(T)));
-            for (size_t i = 0; i < m_size; ++i)
-                new (&m_elements[i]) T();
-        }
+        if (size == 0)
+            return FixedArray<T>();
+        T* elements = static_cast<T*>(kmalloc_array(size, sizeof(T)));
+        if (!elements)
+            return Error::from_errno(ENOMEM);
+        for (size_t i = 0; i < size; ++i)
+            new (&elements[i]) T();
+        return FixedArray<T>(size, elements);
     }
-    ~FixedArray()
+
+    static FixedArray<T> must_create_but_fixme_should_propagate_errors(size_t size)
     {
-        clear();
+        return MUST(try_create(size));
     }
 
-    FixedArray(FixedArray const& other)
-        : m_size(other.m_size)
+    ErrorOr<FixedArray<T>> try_clone() const
+    {
+        if (m_size == 0)
+            return FixedArray<T>();
+        T* elements = static_cast<T*>(kmalloc_array(m_size, sizeof(T)));
+        if (!elements)
+            return Error::from_errno(ENOMEM);
+        for (size_t i = 0; i < m_size; ++i)
+            new (&elements[i]) T(m_elements[i]);
+        return FixedArray<T>(m_size, elements);
+    }
+
+    FixedArray<T> must_clone_but_fixme_should_propagate_errors() const
     {
-        if (m_size != 0) {
-            m_elements = static_cast<T*>(kmalloc_array(m_size, sizeof(T)));
-            for (size_t i = 0; i < m_size; ++i)
-                new (&m_elements[i]) T(other[i]);
-        }
+        return MUST(try_clone());
     }
 
-    FixedArray& operator=(FixedArray const& other)
+    //NOTE: Nobody can ever use these functions, since it would be impossible to make them OOM-safe due to their signatures. We just explicitly delete them.
```

Comment:
```suggestion
    // NOTE: Nobody can ever use these functions, since it would be impossible to make them OOM-safe due to their signatures. We just explicitly delete them.
```